### PR TITLE
test: add GitHub action for end-to-end tests

### PR DESCRIPTION
**Pull request by erikgrinaker**
Partial fix for #5291.

### CODE REVIEW

File: test/e2e/docker/Dockerfile
Comment (melekes) [inline test/e2e/docker/Dockerfile:6]: why do u need to suppress the output?

Comment (erikgrinaker) [reply test/e2e/docker/Dockerfile:6]: apt-get outputs a ton of useless noise which makes the CI logs harder to read.

**Review by melekes — APPROVED** [review]
👍

**Review by marbar3778 — APPROVED** [review]
LGTM. If this action becomes required paths will have to be replaced

**Comment by erikgrinaker** [comment]
Yeah, it should be required, will update GitHub after merging.

**Comment by marbar3778** [comment]
> Yeah, it should be required, will update GitHub after merging.

With the current flow the action will sit as pending if none of the paths are touched. I can't recall if this blocks merging or not though. We will want to mimic the flow in the tests.yml file. 

```yml
     - uses: technote-space/get-diff-action@v3
        with:
          SUFFIX_FILTER: |
            .go
            .mod
            .sum
          SET_ENV_NAME_INSERTIONS: 1
          SET_ENV_NAME_LINES: 1
```

**Comment by erikgrinaker** [comment]
Oh, right - yeah, I was wondering why we implemented this in two different ways. I just picked the simplest one. :) Will update.

**Pull request merged**
